app/routes/account.$accountId.$appId.plan/route.tsx

```diff
@@ -23,107 +25,111 @@ export const meta: MetaFunction = () => {
     return redirect(`/account/${accountId}/${params.appId}`)
   }
+
   const user = await requireUser(request)
-  invariant(user.profile.id && user.profile.emails, "user not found")
-  const userId = getPoktId(user.profile.id)
   const portal = initPortalClient({ token: user.accessToken })
 
   try {
-    const { endpoint } = await portal.endpoint({
-      endpointID: appId,
+    const getUserPortalAppResponse = await portal.getUserPortalApp({
+      portalAppID: appId,
     })
-    const uEmail = user?.profile?._json?.email ?? ""
-    const customer = await getCustomer(uEmail, userId)
-
-    if (customer) {
-      const subscriptions = await stripe.subscriptions.list({
-        customer: customer.id,
-      })
-
-      const subscription = subscriptions.data.filter(
-        (sub) => sub.metadata.endpoint_id && sub.metadata.endpoint_id === endpoint.id,
-      )[0]
-
-      if (subscription) {
-        if (subscription.latest_invoice) {
-          const usageRecords = await stripe.subscriptionItems.listUsageRecordSummaries(
-            subscription.items.data[0].id,
-            { limit: 3 },
-          )
-          const invoiceLatest = await stripe.invoices.retrieve(
-            `${subscription.latest_invoice}`,
-          )
-
-          const invoicePeriodStart = dayjs
-            .unix(Number(invoiceLatest.period_start))
-            .toISOString()
-          const invoicePeriodEnd = dayjs
-            .unix(Number(invoiceLatest.period_start))
-            .toISOString()
-          const relaysLatestInvoice = await getRelays(
-            "endpoints",
-            invoicePeriodStart,
-            invoicePeriodEnd,
-            endpoint.id,
-          )
-
-          return json(
-            {
-              error: false,
-              subscription: subscription,
-              usageRecords,
-              invoice: invoiceLatest,
-              relaysLatestInvoice,
-              endpoint,
-            },
-            {
-              headers: {
-                "Cache-Control": `private, max-age=${
-                  process.env.NODE_ENV === "production" ? "3600" : "60"
-                }`,
-              },
-            },
-          )
-        }
+    if (!getUserPortalAppResponse.getUserPortalApp) {
+      throw new Error(`Account ${params.appId} not found for user ${user.portalUserId}`)
+    }
+    const app = getUserPortalAppResponse.getUserPortalApp
+
+    let subscription
+    let usageRecords
+    let latestInvoice
+    let latestInvoiceRelays
+
+    if (app.legacyFields.stripeSubscriptionID) {
+      subscription = await stripe.subscriptions.retrieve(
+        app.legacyFields.stripeSubscriptionID,
+      )
+
+      if (!subscription) {
+        throw new Error(`Subscription not found for app ${app.id}`)
       }
 
-      return redirect(`/account/${accountId}/${params.appId}`)
+      usageRecords = await stripe.subscriptionItems.listUsageRecordSummaries(
+        subscription.items.data[0].id,
```

Comment:
Is `items.data` array always going to be populated with `data`, is it safe to always access it this way ` subscription.items.data[0].id`?

Reply:
yes, on checkout we create a session with one and only one product in it. There currently will never be more than one product. So this is safe